ovs/extensions/healthcheck/expose_to_cli.py

```diff
@@ -0,0 +1,324 @@
+# in the LICENSE.txt file of the Open vStorage OSE distribution.
+#
+# Open vStorage is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY of any kind.
+import ast
+import imp
+import os
+import inspect
+from datetime import datetime, timedelta
+from ovs.extensions.healthcheck.result import HCResults
+from ovs.log.log_handler import LogHandler
+
+
+# class decorator
+# noinspection PyPep8Naming
+class expose_to_cli(object):
+    def __init__(self, module_name=None, method_name=None):
+        if module_name and method_name:
+            self.module_name = module_name
+            self.method_name = method_name
+
+    def __call__(self, func):
+        self.function = func
+        func.module_name = self.module_name
+        func.method_name = self.method_name
+        return func
+    
+    
+class CLIRunner(object):
+    """
+    Runs a method exposed by the expose_to_cli decorator. Serves as a base for all extensions using expose_to_cli
+    """
+    logger = LogHandler.get("ovs", "clirunner")
+    OPTIONAL_ARGUMENTS = ["--help"]
+    START_PATH = ''.join([os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir))])
+
+    def __init__(self):
+        pass
+
+    @classmethod
+    def run_method(cls, *args):
+        """
+        Executes the given method
+        :return:
+        """
+        module_name = ''
+        method_name = ''
+        # Extract option arguments
+        optional_arguments = [arg for arg in cls.OPTIONAL_ARGUMENTS if arg in args]
+        args = [arg for arg in args if arg not in optional_arguments]
+        # Check for remaining arguments
+        if len(args) >= 1:
+            module_name = args[0]
+        if len(args) >= 2:
+            method_name = args[1]
+        # Determine method to execute
+        if (method_name and not module_name) or "--help" in optional_arguments:
+            print "Both the module name and the method name must be specified.".format(method_name, module_name)
+            return cls.print_help(module_name, method_name)
+        # Find the required method and execute it
+        discovered_data = cls._discover_methods(module_name, method_name)
+        if discovered_data is None:
+            print "Found no method {0} for module {1}".format(method_name, module_name)
+            return cls.print_help(module_name)
+        interrupted = False
+        try:
+            for exposed_module, exposed_methods in discovered_data.iteritems():
+                if module_name and exposed_module != module_name:
+                    continue
+                for method_data in exposed_methods:
+                    if not method_name or method_data['method_name'] == method_name:
+                        mod = imp.load_source(method_data['module_name'], method_data['location'])
+                        cl = getattr(mod, method_data['class'])()
+                        if '--help' in optional_arguments:
```

Comment:
I don't understand why this `if` would ever be reached since the `optional_arguments` don't change right between here and line 68. And on line 68 if `--help` is in them, it would have returned there already